python/pyspark/pandas/data_type_ops/num_ops.py

```diff
@@ -181,6 +181,39 @@ class IntegralOps(NumericOps):
     LongType, IntegerType, ByteType and ShortType.
     """
 
+    def __xor__(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
+        _sanitize_list_like(right)
+
+        right_is_boolean = (
+            True
+            if isinstance(right, bool) or (hasattr(right, "dtype") and right.dtype == "bool")
+            else False
+        )
+
+        if isinstance(right, IndexOpsMixin) and isinstance(right.dtype, extension_dtypes):
+            return right.__xor__(left)
+        elif (
```

Comment:
Shall we extract this `elif` condition as a method and reuse it?

Reply:
thanks, deduplicated.